packages/@ember/-internals/runtime/lib/mixins/array.js

```diff
@@ -874,18 +874,27 @@ const ArrayMixin = Mixin.create(Enumerable, {
     - `index` is the current index in the iteration.
     - `array` is the array object itself.
 
-    It must return a truthy value (i.e. `true`) to include an item in the
-    results. Any non-truthy return value will discard the item from the
-    results.
+    A return of a falsey value (i.e. `false`) will continue evaluating items in
```

Comment:
The current paragraph is describing the callback, so it does not make sense to talk about short-circuiting. I also do not think describing it this technically is useful for the average reader.
Maybe you can adjust the description of the callback and say that the result of `any` is true if the callback returns true, otherwise it's false?